CONTRIBUTING.md

```diff
@@ -252,6 +252,71 @@ There may, at times, need to be exceptions where breaking changes are allowed in
 discretion of the project's maintainers, and must be carefully considered before merging. An example of an allowed
 breaking change might be a fix for a behavioral bug that was released in an initial minor version (such as `v0.3.0`).
 
+
+## API conventions
+
+In general we adhere to the [Kubernetes API conventions](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#optional-vs-required).
+We have a small set of exceptions due to the fact that we're using CRDs while the API conventions predates the introduction 
+of CRDs and the way our controllers only patch actual changes to the status instead of updating the entire status in every 
+reconciliation.
+
+### Optional vs. Required
+
+Status fields must be optional, because our controllers are patching selected fields instead of updating the entire status 
+in every reconciliation.
+
+Optional fields have the following properties:
+* In general an optional field must have both an `+optional` marker and an `omitempty` JSON tag.
+* If the semantic difference between nil and the zero value is important for your code, the field must also be a pointer.
+  * Note: This doesn't apply to map or slice types as they already have a built-in `nil` value.
```

Comment:
```suggestion
  * Note: This doesn't apply to map or slice types as they are assignable to `nil`.
```

Reply:
Isn't it rather the other way around `nil` is assignable to map or slice?

Reply:
Either way works, a variable of type map can be assigned `= nil`, and `nil` is assignable to maps and slices

Reply:
I think it's more my understanding of the english language that's the problem :). I'll fix it

Reply:
Feel free to keep it as-is, I reworded to be more concise, not because what you wrote is incorrect